src/main/java/sg/ncl/MainController.java

```diff
@@ -2584,6 +2584,75 @@ public String adminDataManagement(Model model, HttpSession session) {
         return "data_dashboard";
     }
 
+    @RequestMapping("/admin/data/{datasetId}/resources")
+    public String adminViewDataResources(@PathVariable String datasetId, Model model, HttpSession session) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        //----------------------------------------
+        // get list of data resources in a dataset
+        //----------------------------------------
+        Dataset dataset = invokeAndExtractDataInfo(Long.parseLong(datasetId));
+        model.addAttribute("dataset", dataset);
+
+        return "admin_data_resources";
+    }
+
+    @RequestMapping(value = "/admin/data/{datasetId}/resources/{resourceId}/update", method = RequestMethod.GET)
+    public String adminUpdateResource(@PathVariable String datasetId, @PathVariable String resourceId, Model model, HttpSession session) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        Dataset dataset = invokeAndExtractDataInfo(Long.parseLong(datasetId));
+        DataResource currentDataResource = new DataResource();
+
+        for (DataResource dataResource : dataset.getDataResources()) {
+            if (dataResource.getId() == Long.parseLong(resourceId)) {
+                currentDataResource = dataResource;
+                break;
+            }
+        }
+
+        model.addAttribute("did", dataset.getId());
+        model.addAttribute("dataresource", currentDataResource);
+        session.setAttribute("original_dataresource", currentDataResource);
+        return "admin_data_resources_update";
+    }
+
+    @RequestMapping(value = "/admin/data/{datasetId}/resources/{resourceId}/update", method = RequestMethod.POST)
+    public String adminUpdateResourceFormSubmit(@PathVariable String datasetId, @PathVariable String resourceId, @ModelAttribute DataResource dataResource, Model model, HttpSession session, RedirectAttributes redirectAttributes) throws IOException {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        DataResource original = (DataResource) session.getAttribute("original_dataresource");
+        Dataset dataset = invokeAndExtractDataInfo(Long.parseLong(datasetId));
+        Dataset updatedDataset = updateDataset(dataset, dataResource);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "updatedDataset". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "updatedDataset" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)